src/components/main.tsx

```diff
@@ -1,23 +1,40 @@
 import React, { useState, useEffect } from "react";
 import BrightEthereumDeepLinkQR from "./qrGenerator";
-import { Button, Form, Input, Modal, ModalBody, Container, InputGroup, InputGroupAddon } from 'reactstrap';
-import { assignEthereum } from '../util/metamask'
+import { Button, Form, Input, Modal, ModalBody, Container, InputGroup, InputGroupAddon, FormFeedback } from 'reactstrap';
+import { assignEthereum, convertENS } from '../util/metamask'
 import MetaMask from '../assets/img/MetaMask.svg'
 import Web3 from 'web3';
 import { mobileCheck, androidOrIphoneLink } from "util/detectMobile";
 import { deepLinkPrefix } from "util/deepLink";
 
 const Main = () => {
     const [input, updateInput] = useState('');
+    const [resolvedENS, updateResolved] = useState('');
     const [showQR, toggleShowQR] = useState(false);
     const [notSupported, toggleNotSupported] = useState(false);
     const [validAddress, isValid] = useState(false);
     const [isMobile] = useState(mobileCheck())
     const [ethereum] = useState(() => assignEthereum());
     const [brightIdStoreLink] = useState(androidOrIphoneLink());
 
-    const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const handleChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
         updateInput(e.target.value)
+        const {value} = e.target
+        updateInput(value)
+
+        if( !value.includes('.eth') && Web3.utils.isAddress(value) ){
```

Comment:
You are calling `convertENS` on raw addresses that don't need to be converted and handling them in that function. This is not the correct way to handle this. You are performing a whole bunch of extra steps that don't need to be done at all. 

As an example, you could do it like this, rewrote it in a much more succinct way with correct `useEffect` usage and resolving addresses only when necessary: 
```javacript
    const handleChange = async (e: React.ChangeEvent<HTMLInputElement>) => {
        updateInput(e.target.value)
    };

    useEffect(() => {
        if (input.includes('.eth')) {
            async function resolveENS() {
                validateAndUpdateAddress(await convertENS(input))
            }
            resolveENS();
         } else validateAndUpdateAddress(input);
    }, [input])

    const validateAndUpdateAddress = (input: string) => {
        if (Web3.utils.isAddress(input)) {
            isValid(true);
            updateAddress(input);
        } else {
            isValid(false);
        }
    }
```

Reply:
That's fine, you can just target the es6 compilation instead in tsconfig